### DISCUSSION TITLE
ListEvidences returns the evidences even if the user is not allowed

### DISCUSSION THREAD

**Discussion by anatheka** [clouditor/clouditor#1089] [Q&A]
I'm not sure why, but the Evidences (ListEvidences) are returned even if all == false and no filter for the CloudServiceId is given. I think it's because of the following lines [link](https://github.com/clouditor/clouditor/blob/b75643ac8aceffd4550bd5fbc0a46c820d3ae2cc/service/evidence/evidence_store.go#L200):
```
all, allowed = svc.authz.AllowedCloudServices(ctx)
	if !all & req.Filter != nil & req.Filter.CloudServiceId != nil && !slices.Contains(allowed, req.GetFilter().GetCloudServiceId()) {
		return nil, service.ErrPermissionDenied
	}
```

Furthermore, the AllowedCloudServices() fails if the allowed cloud services claim from the token is empty. 
I think we have to rethink the check above.

**Comment by oxisto** [comment]
Ok that would be weird, because we limit the query to the allowed cloud services here.

https://github.com/clouditor/clouditor/blob/b75643ac8aceffd4550bd5fbc0a46c820d3ae2cc/service/evidence/evidence_store.go#L220-L224
